### ISSUE TITLE
Invalid argument error

### ISSUE THREAD

**Issue by eitanaviv** [simolus3/moor#158]
Hi there,
Wanted to start using Moor but encountered errors I couldn't manage to resolve.
pubspec.yaml configuration:
```
dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^0.1.2
  moor_flutter: ^1.7.0

dev_dependencies:
  moor_generator: ^1.7.0
  build_runner:
  flutter_test:
    sdk: flutter
```

Created a file named `tables.dart` with the following code:
```
import 'package:moor_flutter/moor_flutter.dart';

part 'tables.g.dart';

class Todos extends Table {
  IntColumn get id => integer().autoIncrement()();

  TextColumn get content => text()();

  DateTimeColumn get targetDate => dateTime().nullable()();

  IntColumn get category => integer()
      .nullable()
      .customConstraint('NULLABLE REFERENCES categories(id)')();
}

@UseMoor(tables: [Todos])
class MyDatabase {}
```

Getting the following errors after running `flutter packages pub run build_runner build`:
```
[INFO] Generating build script...
[INFO] Generating build script completed, took 336ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 97ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 966ms

[INFO] Running build...
[INFO] Running build completed, took 17ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 55ms

[SEVERE] moor_generator:moor_generator on lib/widget_parser/grid_widget_parser.dart (cached):

Invalid argument(s): Path must be absolute : dart:convert
[SEVERE] moor_generator:moor_generator on lib/widget_parser/textfield_widget_parser.dart (cached):

Invalid argument(s): Path must be absolute : dart:core
[SEVERE] moor_generator:moor_generator on lib/widget_parser/slider_widget_parser.dart (cached):

Invalid argument(s): Path must be absolute : dart:convert
[SEVERE] moor_generator:moor_generator on lib/widget_parser/cached_image_widget_parser.dart (cached):

Invalid argument(s): Path must be absolute : dart:core
[SEVERE] moor_generator:moor_generator on test/widget_test.dart (cached):

Invalid argument(s): Path must be absolute : dart:core
[SEVERE] moor_generator:moor_generator on lib/main.dart (cached):

Invalid argument(s): Path must be absolute : dart:core
[SEVERE] moor_generator:moor_generator on lib/model/string_pair.dart (cached):

Invalid argument(s): Path must be absolute : dart:core
[SEVERE] moor_generator:moor_generator on lib/register_screen.dart (cached):

Invalid argument(s): Path must be absolute : dart:convert
[SEVERE] moor_generator:moor_generator on lib/model/app_data.dart (cached):

Invalid argument(s): Path must be absolute : dart:convert
[SEVERE] moor_generator:moor_generator on lib/db/database_helper.dart (cached):

Invalid argument(s): Path must be absolute : dart:io
[SEVERE] moor_generator:moor_generator on lib/db/tables.dart (cached):

Invalid argument(s): Path must be absolute : dart:async
[SEVERE] moor_generator:moor_generator on lib/webview_screen.dart (cached):

Invalid argument(s): Path must be absolute : dart:core
[SEVERE] moor_generator:moor_generator on lib/remote/api_provider.dart (cached):

Invalid argument(s): Path must be absolute : dart:core
[SEVERE] moor_generator:moor_generator on lib/splash_screen.dart (cached):

Invalid argument(s): Path must be absolute : dart:async
[SEVERE] Failed after 94ms
pub finished with exit code 1
```

It's a new project I just created using Android Studio.
Please advise me on how to fix that.
Thanks!!

**Comment by mikerockett** [comment]
This appears to be due to the analyzer dependency, which you can downgrade to 0.38.2 (from .3) in pubspec:

```yaml
dependencies:
  analyzer: 0.38.2
```

**Comment by simolus3** [comment]
Thanks for trying out moor! @mikerockett is right - it's caused by an analyzer version (https://dartbug.com/38499) and manually downgrading the dependency should fix this problem. I'll close this issue because it's not moor's fault. I'll pin it for better visibility and let you know when the fixed `analyzer` dependency can be removed.

**Comment by simolus3** [comment]
The underlying bug has just been fixed in `build_resolvers: 1.1.0`.

You can remove the fixed `analyzer` dependency from your pubspec and then run `flutter packages upgrade`. The new version will print "[INFO] Generating SDK summary completed" on the first build.
